roles/_change_user_pwd/tasks/main.yml

```diff
@@ -0,0 +1,16 @@
+---
+- name: Change Odoo user password
+  ansible.builtin.expect:
+    command: docker exec -it app_odoo_1 bash -c 'odoo shell -p 8055'
+    responses:
+      '>>> ':
+        # Encrypted login might have newlines, so we omit that.
+        - "login = '{{ user_login | trim }}'"
+        - "user = env['res.users'].search([('login', '=', login)])"
+        - True if user else exit()
+        - "user.password = '{{ user_pwd | trim}}'"
```

Comment:
Shouldn't `new_password` field be used?

Reply:
Well Odoo does same thing when changing via wizard:

```python
    def change_password_button(self):
        for line in self:
            if not line.new_passwd:
                raise UserError(_("Before clicking on 'Change Password', you have to write a new password."))
            line.user_id.write({'password': line.new_passwd})
        # don't keep temporary passwords in the database longer than necessary
        self.write({'new_passwd': False})
```

And in a past I could not use `new_password`, because of:

```
            if user == self.env.user:
                # To change their own password, users must use the client-specific change password wizard,
                # so that the new password is immediately used for further RPC requests, otherwise the user
                # will face unexpected 'Access Denied' exceptions.
                raise UserError(_('Please use the change password wizard (in User Preferences or User menu) to change your own password.'))
```
But no I log in via shell as superuser, so I guess I could do that.

Reply:
I tried it, but actually with new_password, it works incorrectly. It does not change password consistently. Maybe it keeps old password in cache or something that way. I was able to change password by running script 3 times:)

Reply:
Ok. And password is not stored as a plain text in DB?

Reply:
Well you write as plain text and then its encrypted. But its same way whether its via password or new_password.

As a matter of fact, `new_password` does not do anything interesting. It only checks if password is not set by yourself.

Here how password is encrypted:

```python
    def _set_password(self):
        ctx = self._crypt_context()
        hash_password = ctx.hash if hasattr(ctx, 'hash') else ctx.encrypt
        for user in self:
            self._set_encrypted_password(user.id, hash_password(user.password))

    def _set_encrypted_password(self, uid, pw):
        assert self._crypt_context().identify(pw) != 'plaintext'

        self.env.cr.execute(
            'UPDATE res_users SET password=%s WHERE id=%s',
            (pw, uid)
        )
        self.invalidate_cache(['password'], [uid])
```

Reply:
I remember, that in previous odoo versions (not sure which ones), password was stored as a plain text, therefore asking if it's really improved :)

Reply:
This was improved from like version 8 or something. So quite old improvement actually :)